src/main/java/Parser.java

```diff
@@ -0,0 +1,257 @@
+import java.io.IOException;
+import java.util.ArrayList;
+
+/**
+ * Deals with making sense of the user command and formats
+ * the input by the user to a specified format.
+ */
+public class Parser {
+
+	Ui ui;
+	TaskList tasks;
+	DukeReadFile rf;
+	DukeWriteFile wf;
+	boolean isEndLoop;
+
+
+	public Parser(Ui ui, TaskList tasks, DukeReadFile rf, DukeWriteFile wf) {
+
+		this.ui = ui;
+		this.tasks = tasks;
+		this.rf = rf;
+		this.wf = wf;
+		isEndLoop = false;
+	}
+
+
+	/**
+	 * Main bullk of the program where evaluation of the input
+	 * by the user takes place.
+	 *
+	 * @param input Input of the tasks of the user.
+	 */
+	public void evaluate(String input) {
+		int counter = 0;
+		if (input.equals("bye")) {
+			System.out.println(ui.OUTRO);
+			isEndLoop = true;
+		} else if (input.startsWith("find")) {
+			try {
+				if (input.length() < 5) {
+					throw new DukeException(ui.EMPTYINPUT);
+				}
+				String findTask = input.substring(5);
+				System.out.println(ui.BORDER + "\nHere are the matching tasks in your list:");
+				int taskNumber = 1;
+				for(int i = 0; i < tasks.getTaskList().size(); i ++) {
+					if (tasks.getTask(i).toString().contains(findTask)) {
+						System.out.println(taskNumber + "." + tasks.getTask(i).toString());
+						taskNumber++;
+					}
+				}
+				System.out.println(ui.BORDER);
+			} catch (DukeException e) {
```

Comment:
You should probably catch this at a higher scope. There's no point throwing an exception and catching it in the same method.